### ISSUE TITLE
Hibernation?

### ISSUE THREAD

**Issue by Lyndeno** [nix-community/lanzaboote#294]
My understanding is that nixpkgs has Kernel Lockdown disabled. So in theory hibernation should work.

In my case, with lanzaboote, I can hibernate. But my laptop will not resume from the hibernation image, and instead boot fresh.

Can hibernation work with lanzaboote enabled?

**Comment by blitz** [comment]
Given that hibernation is handled by Linux itself, there _should_ be no interaction with Lanzaboote.

Can you share more details about your setup? Does it work with plain systemd-boot without Lanzaboote involved?

**Comment by Lyndeno** [comment]
Before I set up lanzaboote, hibernation was working on this machine. I am also using the experimental systemd initrd and tpm decryption at boot.

**Comment by Lyndeno** [comment]
> Did you try hibernation with the experimental systemd initrd *without* lanzaboote?

I'll double check this one. I think I switched to them both around the same time.

**Comment by Lyndeno** [comment]
I was able to look into this without disabling Lanzaboote. Turns out systemd 255 needed a fix in nixpkgs for resume to work properly and this fix was merged right after I updated my flake inputs, so I updated them again. Resume is fixed in NixOS/nixpkgs#284496. I also had to explicitly set the resumeDevice in my config. I guess that is not automatic with the systemd-initrd. But now Cryptenroll, Lanzaboote, and Hibernation all work together which is very nice.

This was a problem with systemd-initrd, not Lanzaboote. Apologies for the false issue.
